core/src/executor/select/mod.rs

```diff
@@ -151,6 +156,7 @@ pub async fn select_with_labels<'a, T>(
     filter_context: Option<Rc<FilterContext<'a>>>,
     with_labels: bool,
 ) -> Result<(Vec<String>, impl TryStream<Ok = Row, Error = Error> + 'a)> {
+    // Should we add Future here?
```

Comment:
recursion in an `async fn` requires boxing
a recursive `async fn` must be rewritten to return a boxed `dyn Future`
consider using the `async_recursion` crate: https://crates.io/crates/async_recursionrustc[E0733](https://doc.rust-lang.org/error-index.html#E0733)

Reply:
solved with deriving async_recursion